src/components/Table/TableHead/TableHeader.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { settings } from 'carbon-components';
+import {
+  ArrowUp20 as Arrow,
+  ArrowsVertical20 as Arrows,
+} from '@carbon/icons-react';
+
+import { sortStates } from './state/sorting';
+ 
+const { prefix } = settings;
+
+const translationKeys = {
+  iconDescription: 'carbon.table.header.icon.description',
+};
+
+const translateWithId = (key, { sortDirection, isSortHeader }) => {
+  if (key === translationKeys.iconDescription) {
+    if (isSortHeader) {
+      // When transitioning, we know that the sequence of states is as follows:
+      // NONE -> ASC -> DESC -> NONE
+      if (sortDirection === sortStates.NONE) {
+        return 'Sort rows by this header in ascending order';
+      }
+      if (sortDirection === sortStates.ASC) {
+        return 'Sort rows by this header in descending order';
+      }
+
+      return 'Unsort rows by this header';
+    }
+    return 'Sort rows by this header in ascending order';
+  }
+
+  return '';
+};
+
+const sortDirections = {
+  [sortStates.NONE]: 'none',
+  [sortStates.ASC]: 'ascending',
+  [sortStates.DESC]: 'descending',
+};
+
+const TableHeader = React.forwardRef(function TableHeader(
+  {
+    className: headerClassName,
+    children,
+    isSortable,
+    isSortHeader,
+    onClick,
+    scope,
+    sortDirection,
+    translateWithId: t,
+    ...rest
+  },
+  ref
+) {
+  if (!isSortable) {
+    return (
+      <th {...rest} className={headerClassName} scope={scope}   ref={ref}>
+        <span className={`${prefix}--table-header-label`}>{children}</span>
+      </th>
+    );
+  }
+
+  const className = cx(headerClassName, {
+    [`${prefix}--table-sort`]: true,
+    [`${prefix}--table-sort--active`]:
+      isSortHeader && sortDirection !== sortStates.NONE,
+    [`${prefix}--table-sort--ascending`]:
+      isSortHeader && sortDirection === sortStates.DESC,
+  });
+  const ariaSort = !isSortHeader ? 'none' : sortDirections[sortDirection];
+
+  return (
+    <th
+      scope={scope}
+      className={headerClassName}
+      aria-sort={ariaSort}
+      ref={ref} 
+      {...rest} >
```

Comment:
...rest property is moved in <th> from inner wrapper, because style(width) was not applying on <th> while resizing. This also need to be updated in Carbon component.

Reply:
@davidicus Can you please deliver this Carbon changes, I think i don't have access on Carbon.

Reply:
Carbon is an open source project. Everyone has access to open a PR. I will see about pushing this change up though.

Reply:
@davidicus @stuckless 
I cloned the carbon branch (https://github.com/carbon-design-system/carbon) for this additional changes in TableHeader.js (adding ref for sorting <th> and moving ...rest property from button to <th>).
So I did the changes in TableHeader.js and executed yarn and yarn test with snapshot update
Also created a issue #5047 in Carbon (https://github.com/carbon-design-system/carbon/issues/5047) 
and commited the changes in Branch: 3715-Datatable-Column-Resize
but unable to push the changes (git push origin 3715-Datatable-Column-Resize) due to access permission. 
Error is : remote: Permission to carbon-design-system/carbon.git denied to sunilkotiyaibm.
Please help on this.
Looping @stuckless @JordanWSmith15

Reply:
@davidicus  I cloned the carbon branch (https://github.com/carbon-design-system/carbon) for this additional changes in TableHeader.js (adding ref for sorting and moving ...rest property from button to ).
So I did the changes in TableHeader.js and executed yarn and yarn test with snapshot update
Also created a issue #5047 in Carbon (carbon-design-system/carbon#5047)
and commited the changes in Branch: 3715-Datatable-Column-Resize
but unable to push the changes (git push origin 3715-Datatable-Column-Resize) due to access permission.
Error is : remote: Permission to carbon-design-system/carbon.git denied to sunilkotiyaibm.
Please help on this.
Looping @stuckless @JordanWSmith15 @psdixit

Reply:
The Carbon repo utilizes the forked git flow. If you look at their [contribution guidelines](https://github.com/carbon-design-system/carbon/blob/master/.github/CONTRIBUTING.md ) you can get guidance. It requires that you fork the Carbon repo and push your changes to your fork. Once this is done you will be able to open a PR on the upstream repository.